datastore/tests/unit/test_transaction.py

```diff
@@ -212,6 +245,95 @@ class Foo(Exception):
         self.assertIsNone(xact.id)
         self.assertEqual(ds_api.begin_transaction.call_count, 1)
 
+    def test_default(self):
+        project = 'PROJECT'
+        id_ = 850302
+        ds_api = _make_datastore_api(xact=id_)
+        client = _Client(project, datastore_api=ds_api)
+        xact = self._make_one(client)
+        self.assertTrue(isinstance(xact.options, self._get_options_class()))
+
+    def test_read_only(self):
+        project = 'PROJECT'
+        id_ = 850302
+        ds_api = _make_datastore_api(xact=id_)
+        client = _Client(project, datastore_api=ds_api)
+        options = self._make_options(read_only=self._make_options().ReadOnly())
+        xact = self._make_one(client, options=options)
+        xact.begin()
+        begin = ds_api.begin_transaction
+        begin.assert_called_once()
+        self.assertEqual(begin.call_count, 1)
+
+    def test_read_write(self):
+        project = 'PROJECT'
+        id_ = 850304
+        ds_api = _make_datastore_api(xact_id=id_)
+        client = _Client(project, datastore_api=ds_api)
+        options = self._make_options(read_write=
+                                     self._make_options().ReadWrite())
+        xact = self._make_one(client, options=options)
+        xact.begin()
+        begin = ds_api.begin_transaction
+        begin.assert_called_once()
+        self.assertEqual(begin.call_count, 1)
+
+    def test_previous_tid(self):
+        project = 'PROJECT'
+        id_ = 943243
+        ds_api = _make_datastore_api(xact_id=id_)
+        client = _Client(project, datastore_api=ds_api)
+        options = self._make_options(read_write=
+                                     self._make_options().ReadWrite(
+                                         previous_transaction=b'321'))
+        xact = self._make_one(client, options=options)
+        xact.begin()
+        begin = ds_api.begin_transaction
+        begin.assert_called_once()
+        self.assertEqual(begin.call_count, 1)
+
+    def test_put_read_only(self):
+        project = 'PROJECT'
+        id_ = 943243
+        ds_api = _make_datastore_api(xact_id=id_)
+        client = _Client(project, datastore_api=ds_api)
+        options = self._make_options(read_only=self._make_options().ReadOnly())
+        entity = _Entity()
+        xact = self._make_one(client, options=options)
+        xact.begin()
+        with self.assertRaises(RuntimeError):
+            xact.put(entity)
+
+    def test_put_read_write(self):
+        project = 'PROJECT'
+        id_ = 943243
+        ds_api = _make_datastore_api(xact_id=id_)
+        client = _Client(project, datastore_api=ds_api)
+        options = self._make_options(read_write=
+                                     self._make_options().ReadWrite(
+                                        previous_transaction=b'321'))
```

Comment:
@chemelnucfin Ditto here on the indents. Also **never** indent after an `=` sign.